### ISSUE TITLE
[Bug]: After pasting HTML content with <a> tags link marks are created in wrong locations

### ISSUE THREAD

**Issue by romansp** [ueberdosis/tiptap#4769]
### Which packages did you experience the bug in?

extension-link

### What Tiptap version are you using?

2.2.1

### What’s the bug you are facing?

I'm trying to paste HTML content from another web page into tiptap.

1. Go to https://templates.tiptap.dev/
2. Don't clear the template document and paste the following HTML content with links in the middle of the doc.

--- start copying below ---

Some search engines:
- [Google](https://google.com/)
- [DuckDuckGo](https://duckduckgo.com/)
- [Bing](https://www.bing.com/)

What is a search engine: https://en.wikipedia.org/wiki/Search_engine

--- stop copying above ---

3. Check pasted content and inspect browser console logs.

### What browser are you using?

Chrome

### Code example

https://stackblitz.com/edit/vue3-vite-starter-pj5vfd

### What did you expect to happen?

Content should be pasted with links parsed correctly and no console errors should be thrown.

### Anything to add? (optional)

<details><summary>Stack trace</summary>
<p>

```
index.js:87 
 Uncaught TypeError: Cannot read properties of undefined (reading 'nodeSize')
    at _Fragment.nodesBetween (index.js:87:60)
    at _Node.nodesBetween (index.js:1212:22)
    at getMarksBetween (getMarksBetween.ts:28:9)
    at PasteRule.handler (markPasteRule.ts:42:31)
    at PasteRule.ts:137:28
    at Array.forEach (<anonymous>)
    at PasteRule.ts:125:13
    at _Fragment.nodesBetween (index.js:88:31)
    at _Node.nodesBetween (index.js:1212:22)
    at run (PasteRule.ts:114:13)
nodesBetween	@	index.js:87
nodesBetween	@	index.js:1212
getMarksBetween	@	getMarksBetween.ts:28
handler	@	markPasteRule.ts:42
(anonymous)	@	PasteRule.ts:137
(anonymous)	@	PasteRule.ts:125
nodesBetween	@	index.js:88
nodesBetween	@	index.js:1212
run	@	PasteRule.ts:114
appendTransaction	@	PasteRule.ts:235
applyTransaction	@	index.js:808
apply	@	index.js:773
dispatchTransaction	@	Editor.ts:349
dispatch	@	index.js:5406
doPaste	@	index.js:3437
editHandlers.paste	@	index.js:3449
view.dom.addEventListener.view.input.eventHandlers.<computed>	@	index.js:2900
```

</p>
</details> 

Introduced in tiptap v2.1.14, most likely with this PR https://github.com/ueberdosis/tiptap/pull/4700.

I'm not sure if parsing HTML content from `PasteEvent` even required here at all, `index` values from `html` won't match locations in `text`. I assume that's why exception is thrown.
https://github.com/ueberdosis/tiptap/blob/8ba30f30e513065151b06f5faa3c572389728875/packages/extension-link/src/link.ts#L170-L185

By that point HTML content and anchor tags were already parsed by prosemirror via `parseHTML` and schema attributes.

### Did you update your dependencies?

- [X] Yes, I’ve updated my dependencies to use the latest version of all packages.

### Are you sponsoring us?

- [ ] Yes, I’m a sponsor. 💖

**Comment by galdeguer-ut** [comment]
Same problem here, it seems related with #4700

**Comment by davidliudev** [comment]
Same issue here.

**Comment by romansp** [comment]
This is now reproducible on https://templates.tiptap.dev/ with the HTML content example from reproduction steps. 

If HTML is pasted in the middle of the document link marks are mismatched and set in incorrect locations.

**Comment by galdeguer-ut** [comment]
Same here, I still see the `nodeSize` error pasting content with links:

<img width="1476" alt="Captura de pantalla 2024-02-01 a las 9 45 16" src="https://github.com/ueberdosis/tiptap/assets/38070872/7787cdee-de06-4373-a70e-820643c379a4">

It throws the exception in the `prosemirror-model` package, [here](https://github.com/ProseMirror/prosemirror-model/blob/master/src/fragment.ts#L34).

**Comment by nk11dev** [comment]
Same problem: `Cannot read properties of undefined (reading 'nodeSize')`.

I noticed that this error occurs when pasting content that has lists with links. Otherwise, when only the link is copied, this error does not appear to me.

**Comment by jakebarshickf1nx** [comment]
I'm still experiencing many issues with pasting content with links in anything after version 2.1.10

**Comment by HananoshikaYomaru** [comment]
this issue is not fixed

![CleanShot 2024-02-10 at 07 38 46@2x](https://github.com/ueberdosis/tiptap/assets/43137033/8145846b-3986-4b0e-9cfb-1b0f9ee61077)


using tiptap 2.2.2

**Comment by romansp** [comment]
After some initial testing this appears fixed in 2.2.3. Thanks!
